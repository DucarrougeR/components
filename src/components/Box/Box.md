```js noeditor
const StatusAndResources = require('../../../styleguide_components/StatusAndResources').StatusAndResources;
<StatusAndResources
  status="experimental"
  githubURL="https://github.com/looker/relens/blob/master/src/components/Box/Box.tsx"
  feedbackTitle="Box Component Feedback" />
```

### About Box

`<Box />` is a low level component whose primary purpose is to provide a utility for easily controlling the spacing around elements in a consistent way using the [Lens spacing values](/#!/Spacing).

***

### Spacing with Box

The `<Box />` component allows you to control it's margin and padding properties for a chosen side.

Using spacing on a `<Box />` is like following a recipe, you select:
```js noeditor
<List type="number">
  <ListItem>The type of spacing, margin or padding</ListItem>
  <ListItem>The side to apply it to </ListItem>
  <ListItem>The amount of space to apply</ListItem>
</List>
```
```js noeditor
const BoxSpacingRecipeTableRender = require('../../../styleguide_components/BoxSpacingRecipeTable').BoxSpacingRecipeTableRender;
<Box my="xxlarge">
  <BoxSpacingRecipeTableRender />
</Box>
```
To get the correct spacing you choose one value from each of the above columns. For example, to get 8px of margin on the right side you would do. `<Box mr="small">` or if you wanted 40px of padding on all sides you would do `<Box p="xxlarge">`.

You can combine different spacing recipes to apply spacing to multiple sides at one time. For example if you wanted 16px of margin on the left and 20px of padding on the top you would do `<Box ml="medium" pt="large">`

#### Spacing Examples
Below are a few more examples of building a spacing recipe to help clarify how to use spacing with the `<Box>` component.
```js
  <Box m="large">Margin All Sides - Large</Box>
  <Box pl="xsmall">Padding left - Extra Small</Box>
  <Box pl="medium">Padding left -  Medium</Box>
  <Box my="xxxlarge" ml="xlarge">Margin Top & Bottom - XXXLarge, Margin left - XLarge</Box>
  <Box pb="xxlarge" ml="xxxlarge"> Padding Bottom - XXLarge, Margin left - XXXLarge</Box>
```

#### Spacing Responsively
Any one of the spacing properties can be passed an array of spacing sizes that will then apply at different breakpoints. There are 5 [breakpoints](/#!/Breakpoints) slots that you can target. Here is an example of what that looks like.
```js
  <Box pl={[
          'xsmall',   // xsmall spacing up to the first breakpoint
          'medium',   // medium spacing after first breakpoint
          'xlarge',   // xlarge spacing after second breakpoint
          'xxxlarge'  // xxxlarge spacing after third breakpoint
         ]}>
      My padding on the left changes with breakpoints
  </Box>
```
