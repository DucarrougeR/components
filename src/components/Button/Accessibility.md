# Accessibility



## Best Practices
- Buttons **do not** open a new page, use links instead
- Buttons can open dialogs and menus, but should use the proper ARIA properties

## Labeling Expectations
- Use clear and descriptive labels. By default this is the text inside the `<Button>` component.
- Ambiguous labels like "click here", "run", "save" are ok for sighted users but can lack context for users using assistive technology. Use the `aria-label` attribute to provide an accessible label for all `<Button>` components with potentially ambiguous labels.
- Icon only buttons must provide a descriptive label using the `aria-label` attribute.

## Focus Expectations

- Buttons with `:focus` should have an outline and/or a visible change to show it has focus/

## Keyboard Expectations

- Buttons with focus should be activated with user hits the `ENTER` and `SPACEBAR` keys
- If button has focus, `TAB` and `SHIFT+TAB` should move focus to next focusable element.

## Aria Properties

### `aria-label`
Use this to provide an accessible label for `<Button>` components. Good for buttons with ambiguous labels and icon buttons

For example `<Button aria-label="Save Dashboard">Save</Button>` would be announced as "Save Dashboard Button" by a screen reader instead of just "Save Button"

### `aria-haspopup`
Use this to indicate that when the buttons is pressed a popup is triggered. A popup can be a dialog or menu.

For example clicking an icon buttons to reveal a set of options in a context menu you do the following

```
<IconGearButton aria-haspopup="true">
```

If a button opens a modal dialog window you would the following

```
<Button aria-haspopup="dialog">New Schedule</Button>
```


