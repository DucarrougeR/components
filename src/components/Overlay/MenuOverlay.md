```js noeditor
const StatusAndResources = require('../../../styleguide_components/StatusAndResources')
  .StatusAndResources
;<StatusAndResources
  status="stable"
  githubURL="https://github.com/looker/lens/blob/master/src/components/Overlays/MenuOverlay.tsx"
  feedbackTitle="Popover Component Feedback"
/>
```

### Menu Overlay

The `MenuOverlay` component is composed of an `Overlay` with a transparent backdrop. The placement of the backdrop is also customizable.

Try scrolling down so the two menu buttons are at the top of the screen. The backdrop is adjusted 50px from the top, so clicking the other menu button while the current menu overlay is still open will close the current menu and open the new menu overlay.

```js
const Menus = () => {
  const content = (<Box width="260px" maxHeight="90vh" my="none" style={{ overflowY: 'auto' }}>
    <Text>Menu Content</Text>
    <List>
      <ListItem>Thing 1</ListItem>
      <ListItem>Thing 2</ListItem>
    </List>
  </Box>
  )
  return (
  <Flex justifyContent="space-evenly">
    <MenuOverlay placement="bottom" backdrop={{ top: '50px' }} content={content}>
      <Button>Menu One</Button>
    </MenuOverlay>
    <MenuOverlay placement="bottom" backdrop={{ top: '50px' }} content={content}>
      <Button>Menu Two</Button>
    </MenuOverlay>
  </Flex>
  )

}

<Menus />
```
