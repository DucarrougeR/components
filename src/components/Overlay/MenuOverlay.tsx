import * as React from 'react'
import { palette, shadows } from '../../style'
import { radii } from '../../style/radii'
import { CustomizableAttributes } from '../../types/attributes'
import { ManagedModalProps, ModalSurfaceStyleProps } from '../Modal'
import { ModalManager, ModalManagerProps } from '../Modal/ModalManager'
import {
  Overlay,
  OverlayChildrenProps,
  OverlayInteractiveProps,
  OverlaySurface,
} from './'

export interface MenuOverlayInternalProps extends OverlayInteractiveProps {
  arrow?: boolean
  backdropOffset?: {
    top?: string
    left?: string
    bottom?: string
    right?: string
  }
}

const MenuOverlayInternal: React.FC<MenuOverlayInternalProps> = ({
  arrow = true,
  children,
  backdropOffset,
  ...overlayProps
}) => {
  const backdrop = {
    background: 'transparent',
    ...backdropOffset,
  }

  return (
    <Overlay
      backdrop={backdrop}
      experimentalModifiers={{ flip: { enabled: false } }}
      {...overlayProps}
    >
      {(props: OverlayChildrenProps) => (
        <OverlaySurface
          arrow={arrow}
          {...props}
          {...CustomizableMenuOverlayAttributes.surface}
        >
          {children}
        </OverlaySurface>
      )}
    </Overlay>
  )
}

export interface MenuOverlayProps extends ModalManagerProps {
  arrow?: boolean
  backdropOffset?: {
    top?: string
    left?: string
    bottom?: string
    right?: string
  }
}

export class MenuOverlay extends ModalManager<MenuOverlayProps> {
  constructor(props: MenuOverlayProps) {
    super(props)

    this.checkClickOrigin = this.checkClickOrigin.bind(this)
  }

  public open() {
    this.setState(prevState => ({ isOpen: !prevState.isOpen }))
    window.addEventListener('keydown', this.handleEscapePress)
    document.addEventListener('mousedown', this.checkClickOrigin)
  }

  public checkClickOrigin(event: MouseEvent) {
    if (
      this.portalRef.current &&
      this.portalRef.current.contains(event.target as Node)
    ) {
      return
    }

    if (this.triggerRef.current) {
      if (!this.triggerRef.current.contains(event.target as Node)) {
        this.close()
      }
    }
  }

  protected renderModal(content: string, props: ManagedModalProps) {
    return (
      <MenuOverlayInternal
        isOpen={this.state.isOpen}
        triggerRef={this.triggerRef}
        onClose={this.close}
        {...props}
      >
        {content}
      </MenuOverlayInternal>
    )
  }
}

export interface CustomizableMenuOverlayAttributes
  extends CustomizableAttributes {
  surface: ModalSurfaceStyleProps
}

export const CustomizableMenuOverlayAttributes: CustomizableMenuOverlayAttributes = {
  surface: {
    backgroundColor: palette.white,
    border: '1px solid',
    borderColor: palette.charcoal200,
    borderRadius: radii.medium,
    boxShadow: shadows[3],
    color: palette.charcoal900,
  },
}
