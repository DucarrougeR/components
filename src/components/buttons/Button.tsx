import * as React from 'react'
import * as classNames from 'classnames'
import * as styles from './buttons.scss'
import { Button as SButton } from 'semantic-ui-react'

interface ButtonProps {
  className?: string
  danger?: boolean
  onClick?: (...args: any[]) => void
}

export const Button: React.StatelessComponent<ButtonProps> = ({children, className, danger, onClick}) => {
  const btnClass = classNames(
    styles.btn,
    className,
    {
      [styles.btnDanger]: danger
    })

  return <SButton className={btnClass} onClick={onClick ? onClick : () => {}}>{children}</SButton>
}

