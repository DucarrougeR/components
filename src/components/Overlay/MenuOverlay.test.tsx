import 'jest-styled-components'
import * as React from 'react'
import { assertSnapshot } from '../../../test/utils/snapshot'
import { Box } from '../Box'
import { Button } from '../Button'
import { MenuOverlay, MenuOverlayProps } from './MenuOverlay'

const content = <Box>Hello World</Box>

const TestMenuOverlay: React.SFC<MenuOverlayProps> = ({ ...props }) => (
  <MenuOverlay {...props}>
    <Button>Trigger</Button>
  </MenuOverlay>
)

describe('MenuOverlay', () => {
  test('Normal menu overlay', () => {
    assertSnapshot(<TestMenuOverlay content={content} />)
  })

  test('Open menu overlay with backdrop top=50px', () => {
    assertSnapshot(
      <TestMenuOverlay content={content} backdropTop="50px" open />
    )
  })

  test('Open menu overlay with backdrop bottom=50px', () => {
    assertSnapshot(
      <TestMenuOverlay content={content} backdropBottom="50px" open />
    )
  })

  test('Open menu overlay with backdrop left=50px', () => {
    assertSnapshot(
      <TestMenuOverlay content={content} backdropLeft="50px" open />
    )
  })

  test('Open menu overlay with backdrop right=50px', () => {
    assertSnapshot(
      <TestMenuOverlay content={content} backdropRight="50px" open />
    )
  })
})
