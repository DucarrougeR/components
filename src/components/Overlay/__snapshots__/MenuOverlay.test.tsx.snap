// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuOverlay Normal menu overlay 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: inline-block;
  position: relative;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c1 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  padding-left: 1rem;
  padding-right: 1rem;
  font-size: 1rem;
  line-height: 34px;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #473764;
  border-color: #473764;
}

.c1:active,
.c1.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

<div
  className="c0"
>
  <button
    className="c1 c2"
    onClick={[Function]}
  >
    Trigger
  </button>
</div>
`;

exports[`MenuOverlay Open menu overlay with backdrop bottom=50px 1`] = `
Array [
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  bottom: 0;
  background-color: #DEE1E5;
  left: 0;
  opacity: 0.6;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

<div
    className="c0"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": "50px",
        "cursor": "default",
        "left": undefined,
        "opacity": 0,
        "right": undefined,
        "top": undefined,
      }
    }
  />,
  .c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: inline-block;
  position: relative;
  z-index: 1;
}

.c1 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  padding-left: 1rem;
  padding-right: 1rem;
  font-size: 1rem;
  line-height: 34px;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #473764;
  border-color: #473764;
}

.c1:active,
.c1.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

<div
    className="c0"
  >
    <button
      className="c1 c2"
      onClick={[Function]}
    >
      Trigger
    </button>
  </div>,
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  z-index: 1;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: 1px solid;
  border-radius: 0.25rem;
  border-color: #DEE1E5;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  color: #262D33;
  background-color: #FFFFFF;
  margin: 0.5rem;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c3 {
  position: absolute;
}

.c3::before {
  content: '';
  margin: auto;
  display: block;
  width: 0.5rem;
  height: 0.5rem;
  background: #FFFFFF;
  border-bottom: 1px solid #DEE1E5;
  border-right: 1px solid #DEE1E5;
}

.c3[data-placement*='top'] {
  bottom: 0.25rem;
  margin: 0 1rem;
}

.c3[data-placement*='top']::before {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c3[data-placement*='right'] {
  left: 0.25rem;
  margin: 1rem 0;
}

.c3[data-placement*='right']::before {
  -webkit-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
}

.c3[data-placement*='bottom'] {
  top: 0.25rem;
  margin: 0 1rem;
}

.c3[data-placement*='bottom']::before {
  -webkit-transform: rotate(225deg);
  -ms-transform: rotate(225deg);
  transform: rotate(225deg);
}

.c3[data-placement*='left'] {
  right: 0.25rem;
  margin: 1rem 0;
}

.c3[data-placement*='left']::before {
  -webkit-transform: rotate(315deg);
  -ms-transform: rotate(315deg);
  transform: rotate(315deg);
}

<div
    className="c0"
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        Hello World
      </div>
      <div
        className="c3"
        style={Object {}}
      />
    </div>
  </div>,
]
`;

exports[`MenuOverlay Open menu overlay with backdrop left=50px 1`] = `
Array [
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  bottom: 0;
  background-color: #DEE1E5;
  left: 0;
  opacity: 0.6;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

<div
    className="c0"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": undefined,
        "cursor": "default",
        "left": "50px",
        "opacity": 0,
        "right": undefined,
        "top": undefined,
      }
    }
  />,
  .c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: inline-block;
  position: relative;
  z-index: 1;
}

.c1 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  padding-left: 1rem;
  padding-right: 1rem;
  font-size: 1rem;
  line-height: 34px;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #473764;
  border-color: #473764;
}

.c1:active,
.c1.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

<div
    className="c0"
  >
    <button
      className="c1 c2"
      onClick={[Function]}
    >
      Trigger
    </button>
  </div>,
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  z-index: 1;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: 1px solid;
  border-radius: 0.25rem;
  border-color: #DEE1E5;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  color: #262D33;
  background-color: #FFFFFF;
  margin: 0.5rem;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c3 {
  position: absolute;
}

.c3::before {
  content: '';
  margin: auto;
  display: block;
  width: 0.5rem;
  height: 0.5rem;
  background: #FFFFFF;
  border-bottom: 1px solid #DEE1E5;
  border-right: 1px solid #DEE1E5;
}

.c3[data-placement*='top'] {
  bottom: 0.25rem;
  margin: 0 1rem;
}

.c3[data-placement*='top']::before {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c3[data-placement*='right'] {
  left: 0.25rem;
  margin: 1rem 0;
}

.c3[data-placement*='right']::before {
  -webkit-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
}

.c3[data-placement*='bottom'] {
  top: 0.25rem;
  margin: 0 1rem;
}

.c3[data-placement*='bottom']::before {
  -webkit-transform: rotate(225deg);
  -ms-transform: rotate(225deg);
  transform: rotate(225deg);
}

.c3[data-placement*='left'] {
  right: 0.25rem;
  margin: 1rem 0;
}

.c3[data-placement*='left']::before {
  -webkit-transform: rotate(315deg);
  -ms-transform: rotate(315deg);
  transform: rotate(315deg);
}

<div
    className="c0"
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        Hello World
      </div>
      <div
        className="c3"
        style={Object {}}
      />
    </div>
  </div>,
]
`;

exports[`MenuOverlay Open menu overlay with backdrop right=50px 1`] = `
Array [
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  bottom: 0;
  background-color: #DEE1E5;
  left: 0;
  opacity: 0.6;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

<div
    className="c0"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": undefined,
        "cursor": "default",
        "left": undefined,
        "opacity": 0,
        "right": "50px",
        "top": undefined,
      }
    }
  />,
  .c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: inline-block;
  position: relative;
  z-index: 1;
}

.c1 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  padding-left: 1rem;
  padding-right: 1rem;
  font-size: 1rem;
  line-height: 34px;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #473764;
  border-color: #473764;
}

.c1:active,
.c1.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

<div
    className="c0"
  >
    <button
      className="c1 c2"
      onClick={[Function]}
    >
      Trigger
    </button>
  </div>,
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  z-index: 1;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: 1px solid;
  border-radius: 0.25rem;
  border-color: #DEE1E5;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  color: #262D33;
  background-color: #FFFFFF;
  margin: 0.5rem;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c3 {
  position: absolute;
}

.c3::before {
  content: '';
  margin: auto;
  display: block;
  width: 0.5rem;
  height: 0.5rem;
  background: #FFFFFF;
  border-bottom: 1px solid #DEE1E5;
  border-right: 1px solid #DEE1E5;
}

.c3[data-placement*='top'] {
  bottom: 0.25rem;
  margin: 0 1rem;
}

.c3[data-placement*='top']::before {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c3[data-placement*='right'] {
  left: 0.25rem;
  margin: 1rem 0;
}

.c3[data-placement*='right']::before {
  -webkit-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
}

.c3[data-placement*='bottom'] {
  top: 0.25rem;
  margin: 0 1rem;
}

.c3[data-placement*='bottom']::before {
  -webkit-transform: rotate(225deg);
  -ms-transform: rotate(225deg);
  transform: rotate(225deg);
}

.c3[data-placement*='left'] {
  right: 0.25rem;
  margin: 1rem 0;
}

.c3[data-placement*='left']::before {
  -webkit-transform: rotate(315deg);
  -ms-transform: rotate(315deg);
  transform: rotate(315deg);
}

<div
    className="c0"
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        Hello World
      </div>
      <div
        className="c3"
        style={Object {}}
      />
    </div>
  </div>,
]
`;

exports[`MenuOverlay Open menu overlay with backdrop top=50px 1`] = `
Array [
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  bottom: 0;
  background-color: #DEE1E5;
  left: 0;
  opacity: 0.6;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 1;
}

<div
    className="c0"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": undefined,
        "cursor": "default",
        "left": undefined,
        "opacity": 0,
        "right": undefined,
        "top": "50px",
      }
    }
  />,
  .c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0.25rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  padding-top: 0rem;
  padding-bottom: 0rem;
}

.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  display: inline-block;
  position: relative;
  z-index: 1;
}

.c1 {
  font-weight: 600;
  outline: none;
  -webkit-transition: border 80ms;
  transition: border 80ms;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  vertical-align: middle;
  white-space: nowrap;
  padding-left: 1rem;
  padding-right: 1rem;
  font-size: 1rem;
  line-height: 34px;
  border-style: solid;
  border-width: 0.0625rem;
  background: #64518A;
  border-color: #64518A;
  color: #FFFFFF;
}

.c1:focus {
  box-shadow: 0 0 0 0.15rem rgba(100,81,138,0.25);
}

.c1[disabled] {
  cursor: default;
  -webkit-filter: grayscale(0.3);
  filter: grayscale(0.3);
  opacity: 0.25;
}

.c1[disabled]:hover,
.c1[disabled]:active,
.c1[disabled]:focus {
  background-color: #64518A;
  border-color: #C1C6CC;
}

.c1:hover,
.c1:focus,
.c1.hover {
  background: #473764;
  border-color: #473764;
}

.c1:active,
.c1.active {
  background: #2A1B60;
  border-color: #2A1B60;
}

<div
    className="c0"
  >
    <button
      className="c1 c2"
      onClick={[Function]}
    >
      Trigger
    </button>
  </div>,
  .c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  z-index: 1;
}

.c1 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  border: 1px solid;
  border-radius: 0.25rem;
  border-color: #DEE1E5;
  box-shadow: 0 3px 18px rgba(0,0,0,0.12),0 1px 4px rgba(0,0,0,0.04);
  color: #262D33;
  background-color: #FFFFFF;
  margin: 0.5rem;
}

.c2 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
}

.c3 {
  position: absolute;
}

.c3::before {
  content: '';
  margin: auto;
  display: block;
  width: 0.5rem;
  height: 0.5rem;
  background: #FFFFFF;
  border-bottom: 1px solid #DEE1E5;
  border-right: 1px solid #DEE1E5;
}

.c3[data-placement*='top'] {
  bottom: 0.25rem;
  margin: 0 1rem;
}

.c3[data-placement*='top']::before {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c3[data-placement*='right'] {
  left: 0.25rem;
  margin: 1rem 0;
}

.c3[data-placement*='right']::before {
  -webkit-transform: rotate(135deg);
  -ms-transform: rotate(135deg);
  transform: rotate(135deg);
}

.c3[data-placement*='bottom'] {
  top: 0.25rem;
  margin: 0 1rem;
}

.c3[data-placement*='bottom']::before {
  -webkit-transform: rotate(225deg);
  -ms-transform: rotate(225deg);
  transform: rotate(225deg);
}

.c3[data-placement*='left'] {
  right: 0.25rem;
  margin: 1rem 0;
}

.c3[data-placement*='left']::before {
  -webkit-transform: rotate(315deg);
  -ms-transform: rotate(315deg);
  transform: rotate(315deg);
}

<div
    className="c0"
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "pointerEvents": "none",
        "position": "absolute",
        "top": 0,
      }
    }
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        Hello World
      </div>
      <div
        className="c3"
        style={Object {}}
      />
    </div>
  </div>,
]
`;
